tensorflow/core/kernels/fused_batch_norm_op.cu.cc

```diff
@@ -231,15 +240,18 @@ struct FusedBatchNormInferenceKernel<Eigen::half, float, tensor_format,
                                     /*is_generic_kernel=*/true>;
 
   __device__ static void run(int32 count, int32 channels_size,
-                             int32 inner_dim_size, const T* __restrict__ in,
+                             int32 inner_dim_size, const T* __restrict__ _in,
                              const U* __restrict__ scale,
                              const U* __restrict__ offset,
                              const U* __restrict__ mean,
                              const U* __restrict__ var,
-                             const T* __restrict__ side_input, float epsilon,
-                             T* __restrict__ out) {
+                             const T* __restrict__ _side_input, float epsilon,
+                             T* __restrict__ _out) {
+    const IT* in = reinterpret_cast<const IT*>(_in);
```

Comment:
guessing the renaming + casting is to somehow workaround  the `__restrict__` attribute?
a comment block here explaining why this is needed will be helpful!

Reply:
No, it's just because Eigen::half and __half are two unrelated types. I'm not sure how it works with CUDA.